common/djangoapps/student/views.py

```diff
@@ -1804,6 +1805,28 @@ def create_account_with_params(request, params):
     return new_user
 
 
+def _enroll_user_in_pending_courses(user_id):
+    """
+    Enroll student in any pending courses he/she may have.
+    """
+    student = User.objects.filter(id=user_id)
```

Comment:
I guess I'm just confused why this backend validation is necessary — do you have a compelling reason why it is necessary? It's adding unnecessary cruft (not to mention extra database calls) to pass around a user object, extract its id, then look it up again by its id.